tests/testthat/test-components.R

```diff
@@ -0,0 +1,97 @@
+test_that("predecessor", {
+  predecessor <- get_rendered_standard(
+    "predecessor",
+    list(source = "pharmaverseadam::adsl", by = "USUBJID", variable = "ACTARM")
+  )
+
+  cat(predecessor$template, sep = "\n") |>
+    expect_snapshot()
+
+  advs <- pharmaverseadam::advs |>
+    dplyr::select(USUBJID, PARAMCD, AVAL, ACTARM)
+
+  predecessor$test(
+    input = advs |> dplyr::select(-ACTARM),
+    expected = advs
+  )
+})
+
+test_that("assign", {
+  assign <- get_rendered_standard("assign", list(variable = "y", value = 1))
+
+  cat(assign$template, sep = "\n") |>
+    expect_snapshot()
+
+  df <- data.frame(x = "a", y = 1)
+
+  assign$test(
+    input = df |> dplyr::select(-y),
+    expected = df
+  )
+})
+
+test_that("astdt", {
+  astdt <- get_rendered_standard("astdt", list(dtc = "AESTDTC"))
+
+  cat(astdt$template, sep = "\n") |>
+    expect_snapshot()
+
+  adae <- pharmaverseadam::adae |>
+    dplyr::select(USUBJID, AESTDTC, ASTDT, ASTDTF)
+
+  astdt$test(
+    input = adae |> dplyr::select(-ASTDT, -ASTDTF),
+    expected = adae
+  )
+})
+
+test_that("aendt", {
+  astdt <- get_rendered_standard("aendt", list(dtc = "AEENDTC"))
+
+  cat(astdt$template, sep = "\n") |>
+    expect_snapshot()
+
+  adae <- pharmaverseadam::adae |>
+    dplyr::select(USUBJID, AEENDTC, AENDT, AENDTF)
+
+  astdt$test(
```

Comment:
In the `aendt` test you assign to `astdt` instead of a variable named after the component (e.g. `aendt`), causing confusion and potential test failures.
```suggestion
  aendt <- get_rendered_standard("aendt", list(dtc = "AEENDTC"))

  cat(aendt$template, sep = "\n") |>
    expect_snapshot()

  adae <- pharmaverseadam::adae |>
    dplyr::select(USUBJID, AEENDTC, AENDT, AENDTF)

  aendt$test(
```